### PR TITLE
Linux build fixes and improvements

### PR DESCRIPTION
**Pull request by pixelcluster**
Linux build should work now. I also automated the manual Python script step, so the normal CMake process should be all that's needed to build now.

### CODE REVIEW

File: Source/Shaders/GenerateShaders.py [hunk 1]
```diff
@@ -200,7 +211,7 @@ def main():
 
                 with open(filename, "r") as dpd:
                     for line in dpd:
-                        if "#include" in line:
+                        if "#include \"" in line:
```
Comment (sultim-t) [inline Source/Shaders/GenerateShaders.py]: I wonder what was the failure case?

File: CMakeLists.txt [hunk 2]
```diff
@@ -287,4 +292,11 @@ if (MSVC AND WIN32 AND NOT MSVC_VERSION VERSION_LESS 142)
     target_link_options(RtglExample PRIVATE $<$<CONFIG:Debug>:/INCREMENTAL>)
     target_compile_options(RtglExample PRIVATE $<$<CONFIG:Debug>:/ZI>)
 endif()
-endif()
\ No newline at end of file
+endif()
+
+# Build shaders as custom target
+add_custom_target(Shaders
+                  COMMAND python3 ${CMAKE_CURRENT_SOURCE_DIR}/Source/Shaders/GenerateShaders.py -builddir ${CMAKE_CURRENT_BINARY_DIR}
```
Comment (sultim-t) [inline CMakeLists.txt]: `-builddir` should point to `${CMAKE_CURRENT_BINARY_DIR}/shaders` folder

File: CMakeLists.txt [hunk 3]
```diff
@@ -287,4 +292,11 @@ if (MSVC AND WIN32 AND NOT MSVC_VERSION VERSION_LESS 142)
...
+                  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/Source/Shaders)
+
+add_dependencies(RayTracedGL1 Shaders)
```
Comment (sultim-t) [inline CMakeLists.txt]: For compatibility with CI, and VS Code workflow, can we add an option:

* `option(RG_WITH_SHADERS "Build includes a shader generation" ON)` ?

Then we can specify `-DRG_WITH_SHADERS=OFF` in `.github/workflows/build-windows.yml` for compatibility with CI

**Review by sultim-t — COMMENTED** [review]
Thanks for updating VMA also :)

File: Source/Shaders/GenerateShaders.py (hunk 1 above)
Comment (pixelcluster) [reply Source/Shaders/GenerateShaders.py]: Right, I think we can remove this again. IIRC, what I did wrong was to run the script from the top-level build directory. This picked up C++ headers with `#include <vector>`.

**Pull request merged**